File: sdks/python/bolt_prediction_sdk/models/health_index_factor_schema.py
"""
HealthIndexFactorSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class HealthIndexFactorSchema(SDKBaseModel):
    """健康度因子详情"""

    factor_name: str = Field(description="因子名称")
    factor_code: str = Field(description="因子代码")
    score: float = Field(description="因子得分 0-100")
    weight: float = Field(description="因子权重")
    contribution: float = Field(description="对总健康度的贡献")
    description: Optional[Any] = Field(description="因子描述", default=None)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/HealthIndexFactorSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 健康度因子详情 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class HealthIndexFactorSchema {

    @JsonProperty("factor_name")
    private String factorName;

    @JsonProperty("factor_code")
    private String factorCode;

    @JsonProperty("score")
    private Double score;

    @JsonProperty("weight")
    private Double weight;

    @JsonProperty("contribution")
    private Double contribution;

    @JsonProperty("description")
    private Object description;

    public String getFactorName() {
        return factorName;
    }

    public void setFactorName(String factorName) {
        this.factorName = factorName;
    }

    public String getFactorCode() {
        return factorCode;
    }

    public void setFactorCode(String factorCode) {
        this.factorCode = factorCode;
    }

    public Double getScore() {
        return score;
    }

    public void setScore(Double score) {
        this.score = score;
    }

    public Double getWeight() {
        return weight;
    }

    public void setWeight(Double weight) {
        this.weight = weight;
    }

    public Double getContribution() {
        return contribution;
    }

    public void setContribution(Double contribution) {
        this.contribution = contribution;
    }

    public Object getDescription() {
        return description;
    }

    public void setDescription(Object description) {
        this.description = description;
    }

}